connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDeserializer.scala

```diff
@@ -235,7 +237,7 @@ private[sql] class ProtobufDeserializer(
           writeRecord(new RowUpdater(row), value.asInstanceOf[DynamicMessage])
           updater.set(ordinal, row)
 
-      case (MESSAGE, ArrayType(st: StructType, containsNull)) =>
+      case (MESSAGE, ArrayType(st: DataType, containsNull)) =>
```

Comment:
It's not clear as to why we need to make this change. It does not look like OneOf related. Could you clarify why this would be needed ? Does it cover any specific case ?

Reply:
@mposdev21 Actually, we don't need this here at all; we can add it at the top along with other arraytype types.